pipeline/coaddition.py

```diff
@@ -416,6 +439,11 @@ def _coadd_zogy(
         if bkg_means is None or bkg_sigmas is None:
             raise ValueError('Background must be given if images are not Image objects. ')
 
+        # NOTE -- this will use a LOT of memory.
+        # All of the numpy arrays will now be stored twice -- once in
+        #   the list, and once in the datacube created.  Suggest
+        #   refactoring so that the for loop above just builds the
+        #   datacubes directly instead of going through the lists.
```

Comment:
Have we checked lately how much memory usage we are up to? Is that still a concern limiting our throughput?

Reply:
I'm still worried.  I'd like it if the subtraction pipeline only used 2GB, but that's probably hopeless at this point.  But I'd still like to get it less than 4GB or 5GB.  I haven't looked at it closely yet, but I've idly looked at it and we still go above 5GB, at least in the tests.  (The tests may be more memory-intensive than the actual pipeline, as we keep some extra data around for testing purposes.)

For the coadd pipeline, the memory usage always will be more, because we might be reading in lots of images all at once.  (Turns out swarp can handle not doing that, but also that can make swarp slower as it has more disk i/o.)  So, we won't be able to run as many reference-building pipeline instances on one node at once.  But, that's less of a big deal, because we expect many (many) subtraction runs for each reference built, so it's OK if the reference building is more resource intenstive.  But, still : if we're building a reference from 20 images, having double copies of arrays for all of those 20 images is probably a bit much!  I'm not sure which reference pipeline we're going to use for LS4, the zogy or swarp one.  For decat-wfd, we need to use swarp, because it handles combining together lots of misaligned images better.

At some point in the not-too-distant future, I hope to be really running this on the decat-wfd data, at which point I can look at the memory usage for reference building and subtraction, and start to worry about that for real.